huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go

```diff
@@ -0,0 +1,274 @@
+	return utils.FilterSliceWithField(vaultList, map[string]interface{}{
+		"Billing.ConsistentLevel": d.Get("consistent_level").(string),
+		"Billing.Size":            d.Get("size").(int),
+		"AutoExpand":              d.Get("auto_expand_enabled").(bool),
+	})
+}
+
+func getCbrPolicyOfSpecificVault(client *golangsdk.ServiceClient, vaultId string) (*policies.Policy, error) {
+	opt := policies.ListOpts{
+		VaultID: vaultId,
+	}
+	pages, err := policies.List(client, opt).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	resp, err := policies.ExtractPolicies(pages)
+	if len(resp) > 0 {
+		return &resp[0], nil
+	}
+	return nil, fmtp.Errorf("No policies are bound to the vault.")
+}
+
+func setCbrAllVaultParameters(client *golangsdk.ServiceClient, d *schema.ResourceData,
+	vaultList []interface{}) error {
+	result := make([]map[string]interface{}, len(vaultList))
+	ids := make([]string, len(vaultList))
+	for i, val := range vaultList {
+		vault := val.(vaults.Vault)
+		vMap := map[string]interface{}{
+			"id":                    vault.ID,
+			"name":                  vault.Name,
+			"enterprise_project_id": vault.EnterpriseProjectID,
+			"type":                  vault.Billing.ObjectType,
+			"protection_type":       vault.Billing.ProtectType,
+			"status":                vault.Billing.Status,
+			"consistent_level":      vault.Billing.ConsistentLevel,
+			"size":                  vault.Billing.Size,
+			"allocated":             vault.Billing.Allocated,
+			"used":                  vault.Billing.Used,
+			"spec_code":             vault.Billing.SpecCode,
+			"storage":               vault.Billing.StorageUnit,
+			"auto_expand_enabled":   vault.AutoExpand,
+			"tags":                  utils.TagsToMap(vault.Tags),
+			"resources":             makeCbrVaultResources(vault.Billing.ObjectType, vault.Resources),
+		}
+
+		// Query the CBR policy which bound to the vault by ID
+		if policy, err := getCbrPolicyOfSpecificVault(client, vault.ID); err != nil {
+			logp.Printf("[DEBUG] Unable to find the policy for specific vault: %s", err)
+		} else {
+			vMap["policy_id"] = policy.ID
+		}
+		result[i] = vMap
+		ids[i] = vault.ID
+	}
+
+	d.SetId(hashcode.Strings(ids))
+
+	return d.Set("vaults", result)
+}
+
+func dataSourceCbrVaultsV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.CbrV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating Huaweicloud CBR v3 client: %s", err)
+	}
+
+	opt := buildVaultListOpts(d, config)
+
+	pages, err := vaults.List(client, opt).AllPages()
+	if err != nil {
+		return fmtp.DiagErrorf("Error getting vault details: %s", err)
+	}
+	resp, err := vaults.ExtractVaults(pages)
+
+	// Use the following parameters to filter the result of the List method return: consistent_level, size, auto_expand_enabled.
+	vaultList, err := filterCbrVaults(d, *resp)
+	if err != nil {
+		return fmtp.DiagErrorf("")
```

Comment:
is it not finished yet?

Reply:
That seems to be the case. =-=
The return messsage has been updated.